crates/red_knot_python_semantic/src/semantic_index/use_def.rs

```diff
@@ -278,7 +278,7 @@ mod symbol_state;
 type AllConstraints<'db> = IndexVec<ScopedConstraintId, Constraint<'db>>;
 
 /// Applicable definitions and constraints for every use of a name.
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug, PartialEq, Eq, salsa::Update)]
```

Comment:
I guess maybe it's too late to be worth it now, but it might have made sense to do the `salsa::Update` bound change first in a separate PR, and the coarse-grained structs separately?

I don't _love_ this Salsa-specific trait "infecting" a bunch of previously independent code, but it's pretty harmless and not an invasive change.

Reply:
Ideally yes. But it's not realy possible because of the order in which the PRs merged into Salsa. Coarse grained dependencies reqiures a bug fix that merged after the require `Update` trait PR and adding the `Update` requires the most recent commit which requires fewer `Update` implementations than the first PR...